### PR TITLE
K8SPS-65: Fix updating replication password with semi-sync replicas

### PR DESCRIPTION
**Pull request by egegunes**
The operator restarts replication if the user changes replication
password. If semi-sync replication enabled, the primary is locked due to
semi-sync wait count and orchestrator performs an automatic failover.
With this commit we're disabling semi-sync replication on primary before
restarting replication.

Another issue was making sure that replication is not running before
executing `CHANGE REPLICATION SOURCE TO ...`. It was sporadic but
frequent.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | passed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/51/commits/11cd24283a0df8cfd7a45d54e56ef622110526cc
image: `perconalab/percona-server-mysql-operator:PR-51-11cd242`

File: pkg/replicator/replicator.go [hunk 1]
```diff
@@ -54,7 +55,16 @@ func NewReplicator(user apiv1alpha1.SystemUser, pass, host string, port int32) (
 	return &dbImpl{db}, nil
 }
 
+func (d *dbImpl) StopReplication() error {
+	_, err := d.db.Exec("STOP REPLICA")
+	return err
+}
+
 func (d *dbImpl) ChangeReplicationSource(host, replicaPass string, port int32) error {
+	if err := d.StopReplication(); err != nil {
```
Comment (delgod) [inline pkg/replicator/replicator.go:64]: I would do this explicitly in `reconcileUsers` and `StartReplication` functions

File: controllers/perconaservermysql_controller.go [hunk 2]
```diff
@@ -285,13 +285,53 @@ func (r *PerconaServerMySQLReconciler) reconcileUsers(ctx context.Context, cr *a
...
+					return errors.Wrapf(err, "stop replica %s", hostname)
+				}
+
+				status, _, err := repDb.ReplicationStatus()
+				if err != nil {
+					return errors.Wrapf(err, "get replication status of %s", hostname)
+				}
+
+				for status == replicator.ReplicationStatusActive {
+					time.Sleep(1)
```
Comment (delgod) [inline controllers/perconaservermysql_controller.go:330]: maybe value bigger than 200ms is better

Comment (egegunes) [reply controllers/perconaservermysql_controller.go]: changed to 250ms

File: pkg/replicator/replicator.go (hunk 1 above)
Comment (egegunes) [reply pkg/replicator/replicator.go]: It was unnecessary, deleted.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | passed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/51/commits/40fd6952bc44482e31817e6bac2bf3c6fe6050f0
image: `perconalab/percona-server-mysql-operator:PR-51-40fd695`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| semi-sync | passed |
| monitoring | passed |
| sidecars | passed |
| config | passed |
| users | passed |

commit: https://github.com/percona/percona-server-mysql-operator/pull/51/commits/8f4eb074a5535adfc54727bc9167f01c56521310
image: `perconalab/percona-server-mysql-operator:PR-51-8f4eb07`

**Review by delgod — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**